### PR TITLE
Fix: Record used module

### LINKED ISSUE #132: [Enhancement] Record whether the module has been used in symbol table(Scope)
## Enhancement
To simplify the semantic analysis of the unused check, it is necessary to record whether the module/var has been used in the symbol table(Scope and ScopeObject).

### PR DESCRIPTION
**Pull request by He1pa**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y fix #132 

#### 2. What is the scope of this PR (e.g. component or file name):
sema/resolver/var.rs

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

Fix bug in resolve_var(). When enter child scope,  e.g., in a schema scope, cant find module object.
It should be  recursively search whole scope to lookup module object, not the current scope.element.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by zong-zhe — COMMENTED** [review]
LGTM

File: kclvm/sema/src/resolver/var.rs [hunk 1]
```diff
@@ -84,9 +84,10 @@ impl<'ctx> Resolver<'ctx> {
                 }
             }
         } else {
+            // lookup pkgpath scope and record it as "used"
             if !pkgpath.is_empty() {
-                if let Some(module_scope) = self.scope.borrow_mut().elems.get_mut(pkgpath) {
-                    module_scope.borrow_mut().used = true;
+                if let Some(obj) = self.scope.borrow().lookup(pkgpath) {
```
Comment (Peefy) [inline kclvm/sema/src/resolver/var.rs:89]: Add documents to explain these changed codes.

**Review by zong-zhe — DISMISSED** [review]

**Review by Peefy — APPROVED** [review]
LGTM

**Review by zong-zhe — APPROVED** [review]
LGTM

**Pull request merged**